src/components/zen-dropdown/zen-dropdown.tsx

```diff
@@ -268,8 +268,9 @@ export class ZenDropdown {
     this.opened = this.popover.visible;
   }
 
-  onBeforeToggle(): void {
+  onBeforeToggle(): Promise<void> {
     this.matchDropdownWidth();
+    return new Promise(resolve => resolve());
   }
```

Comment:
```suggestion
  async onBeforeToggle(): Promise<void> {
    this.matchDropdownWidth();
  }
```

1. For an already resolved promise, you could use `return Promise.resolve();`
2. But if you add `async`, the method/function gets automatically wrapped in a promise, and all promises return a new promise, so there's no need to manually return one.